### ISSUE TITLE
Too many concurrent edits to the project configuration - error

### ISSUE THREAD

**Issue by naveensrinivasan** [GoogleCloudPlatform/prometheus-engine#309]
We are trying to use the tube prometheus helm chart for unmanaged workloads https://github.com/GoogleCloudPlatform/prometheus-engine/issues/298 and running into this error.

```
ts=2022-08-23T20:16:48.417Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=200 err="rpc error: code = Aborted desc = Errors during metric descriptor creation: {(metric: prometheus.googleapis.com/net_conntrack_dialer_conn_attempted_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/net_conntrack_dialer_conn_closed_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/net_conntrack_dialer_conn_established_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/net_conntrack_dialer_conn_failed_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_engine_query_duration_seconds_sum/summary:counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_http_request_duration_seconds/histogram, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_http_requests_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_http_response_size_bytes/histogram, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_rule_evaluation_failures_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_rule_evaluations_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_rule_group_iterations_missed_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_rule_group_iterations_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_sd_consul_rpc_duration_seconds_sum/summary:counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_sd_kubernetes_events_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_sd_kubernetes_http_request_duration_seconds_count/summary, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_target_interval_length_seconds/summary, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_target_scrape_pool_sync_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_target_sync_failed_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_target_sync_length_seconds_count/summary, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_target_sync_length_seconds_sum/summary:counter, error: Too many concurrent edits to the project configuration. Please try again.)}."
ts=2022-08-23T20:16:48.891Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=74 err="rpc error: code = Aborted desc = Errors during metric descriptor creation: {(metric: prometheus.googleapis.com/net_conntrack_dialer_conn_attempted_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/net_conntrack_dialer_conn_established_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/net_conntrack_dialer_conn_failed_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_engine_query_duration_seconds_count/summary, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_rule_evaluation_failures_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_rule_evaluations_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_rule_group_iterations_missed_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_sd_kubernetes_events_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_sd_kubernetes_http_request_duration_seconds_sum/summary:counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_target_interval_length_seconds/summary, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_target_scrape_pool_sync_total/counter, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_target_sync_length_seconds_count/summary, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/prometheus_target_sync_length_seconds_sum/summary:counter, error: Too many concurrent edits to the project configuration. Please try again.)}."
ts=2022-08-23T20:17:03.003Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=200 err="rpc error: code = Aborted desc = Errors during metric descriptor creation: {(metric: prometheus.googleapis.com/apiserver_selfrequest_total/counter, error: Too many concurrent edits to the project configuration. Please try again.)}."
ts=2022-08-23T20:17:04.589Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=200 err="rpc error: code = Aborted desc = Errors during metric descriptor creation: {(metric: prometheus.googleapis.com/apiserver_response_sizes/histogram, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/apiserver_selfrequest_total/counter, error: Too many concurrent edits to the project configuration. Please try again.)}."
ts=2022-08-23T20:17:04.591Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=200 err="rpc error: code = Aborted desc = Errors during metric descriptor creation: {(metric: prometheus.googleapis.com/apiserver_response_sizes/histogram, error: Too many concurrent edits to the project configuration. Please try again.), (metric: prometheus.googleapis.com/apiserver_selfrequest_total/counter, error: Too many concurrent edits to the project configuration. Please try again.)}."
ts=2022-08-23T20:17:44.701Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=200 err="rpc error: code = Aborted desc = Errors during metric descriptor creation: {(metric: prometheus.googleapis.com/kube_pod_container_status_waiting_reason/gauge, error: Too many concurrent edits to the project configuration. Please try again.)}."
```

The change in the helm chart for tube-prometheus is 
```
prometheus:
    Container ID:  containerd://15b76a0edbe5f31e1a4177c7130973e5de1469523f1481b1976462a5c6fc4212
    Image:         gke.gcr.io/prometheus-engine/prometheus:v2.35.0-gmp.2-gke.0
    Image ID:      gke.gcr.io/prometheus-engine/prometheus@sha256:02449f32a9fa8ab02c529eb34aa5c13f49d5b6c56f5d3cf23ff7ca23af6eb7d2
    ```
    
    How do we handle this?

**Comment by lyanco** [comment]
See https://cloud.google.com/stackdriver/docs/managed-prometheus/troubleshooting#concurrent-edits

tl;dr: these are transient errors caused by creating a ton of new metrics at once, which we rate limit to protect our system. They should resolve naturally within 30 minutes.

**Comment by naveensrinivasan** [comment]
@lyanco Is this something we need to worry about?

```
ts=2022-08-29T16:47:07.500Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=200 err="rpc error: code = Internal desc = Internal error encountered. Please retry after a few seconds. If internal errors persist, contact support at https://cloud.google.com/support/docs."
ts=2022-08-29T16:47:07.540Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=200 err="rpc error: code = Internal desc = Internal error encountered. Please retry after a few seconds. If internal errors persist, contact support at https://cloud.google.com/support/docs."
ts=2022-08-29T16:47:07.580Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=179 err="rpc error: code = Internal desc = Internal error encountered. Please retry after a few seconds. If internal errors persist, contact support at https://cloud.google.com/support/docs."
ts=2022-08-29T16:47:08.284Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=200 err="rpc error: code = Internal desc = Internal error encountered. Please retry after a few seconds. If internal errors persist, contact support at https://cloud.google.com/support/docs."
ts=2022-08-29T16:47:08.363Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=86 err="rpc error: code = Internal desc = Internal error encountered. Please retry after a few seconds. If internal errors persist, contact support at https://cloud.google.com/support/docs."
ts=2022-08-29T16:47:08.652Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=200 err="rpc error: code = Internal desc = Internal error encountered. Please retry after a few seconds. If internal errors persist, contact support at https://cloud.google.com/support/docs."
ts=2022-08-29T16:47:09.366Z caller=export.go:796 level=error component=gcm_exporter msg="send batch" size=111 err="rpc error: code = Internal desc = Internal error encountered. Please retry after a few seconds. If internal errors persist, contact support at https://cloud.google.com/support/docs."
```

**Comment by lyanco** [comment]
Is this still happening? Or was it transient.
